core/array.rbs

```diff
@@ -1997,6 +1997,10 @@ interface _ToAry[T]
   def to_ary: () -> ::Array[T]
 end
 
+interface _Rand
+  def rand: (::Integer max) -> ::Integer
```

Comment:
Basically the `rand` implementation takes `Integer`, `Float`, `Range`. But focusing to `Array#{shuffle, shuffle!, sample}`, this is minimum restriction, I think.